pkg/scheduler/framework/runtime/framework_test.go

```diff
@@ -0,0 +1,147 @@
+			},
+			isSuccess: false,
+		},
+	}
+
+	registry, err := createAndRegisterFactory(mockFilterPlugin)
+	if err != nil {
+		t.Errorf("create plugin factory error:%v", err)
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			frameWork, err := NewFramework(registry)
+			if err != nil {
+				t.Errorf("create frame work error:%v", err)
+			}
+
+			tt.mockFunc()
+			result := frameWork.RunFilterPlugins(ctx, nil, nil, nil)
+			if result.IsSuccess() != tt.isSuccess {
+				t.Errorf("want %v, but get:%v", tt.isSuccess, result.IsSuccess())
+			}
+		})
+	}
+}
+
+func createAndRegisterFactory(plugin framework.Plugin) (Registry, error) {
+	filterPluginFactory := func() (framework.Plugin, error) {
+		return plugin, nil
+	}
+	registry := Registry{}
+	if err := registry.Register("foo", filterPluginFactory); err != nil {
+		return nil, err
+	}
+
+	return registry, nil
+}
+
+func Test_frameworkImpl_RunScorePlugins(t *testing.T) {
+	ctx := context.Background()
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockScorePlugin := frameworktesting.NewMockScorePlugin(mockCtrl)
+	mockScoreExtension := frameworktesting.NewMockScoreExtensions(mockCtrl)
+	clusters := []*clusterv1alpha1.Cluster{
+		{ObjectMeta: metav1.ObjectMeta{Name: "c1"}},
+	}
+	tests := []struct {
+		name     string
+		mockFunc func()
+		hasErr   bool
+	}{
+		{
+			name: "Test score ok",
+			mockFunc: func() {
+				mockScorePlugin.EXPECT().Score(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+					Return(int64(60), framework.NewResult(framework.Success))
+
+				mockScorePlugin.EXPECT().ScoreExtensions().Times(2).Return(mockScoreExtension)
+				mockScoreExtension.EXPECT().NormalizeScore(gomock.Any(), gomock.Any()).
+					Return(framework.NewResult(framework.Success))
+
+				mockScorePlugin.EXPECT().Name().AnyTimes().Return("foo")
+			},
+			hasErr: false,
+		},
+		{
+			name: "Test score func error",
+			mockFunc: func() {
+				mockScorePlugin.EXPECT().Score(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+					Return(int64(-1), framework.NewResult(framework.Error, "foo"))
+			},
+			hasErr: true,
+		},
+		{
+			name: "Test normalize score error",
+			mockFunc: func() {
+				mockScorePlugin.EXPECT().Score(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).
+					Return(int64(60), framework.NewResult(framework.Success))
+
+				mockScorePlugin.EXPECT().ScoreExtensions().Times(2).Return(mockScoreExtension)
+				mockScoreExtension.EXPECT().NormalizeScore(gomock.Any(), gomock.Any()).
+					Return(framework.NewResult(framework.Error, "foo"))
+			},
+			hasErr: true,
+		},
+	}
+
+	registry, err := createAndRegisterFactory(mockScorePlugin)
+	if err != nil {
+		t.Errorf("create plugin factory error:%v", err)
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			frameWork, err := NewFramework(registry)
+			if err != nil {
+				t.Errorf("create frame work error:%v", err)
+			}
+
+			tt.mockFunc()
```

Comment:
I think every test case should be independent. So it's better not share mock object between test cases. Instead, we can create a new mock object in every test case.

Reply:
It make sense, but I prefer to keep it as case global, it makes code looks more simple

Reply:
mock is polluted between cases, It doesn't seem a good thing.